### ISSUE TITLE
import/named error being incorrectly thrown for victory imports

### ISSUE THREAD

**Issue by kaiyoma** [import-js/eslint-plugin-import#2585]
Valid code (type checking passes and code works at runtime):

```
import { VictoryAxis, VictoryLine, VictoryArea } from 'victory';
```

Errors from eslint:

```
  6:23  error  VictoryAxis not found in 'victory'  import/named
  6:36  error  VictoryLine not found in 'victory'  import/named
  6:49  error  VictoryArea not found in 'victory'  import/named
```

eslint: 8.26.0
eslint-plugin-import: 2.26.0
victory: 36.6.8

**Comment by ljharb** [comment]
Looking at https://unpkg.com/browse/victory@36.6.8/package.json, the "module" and "jsnext:main" fields point to https://unpkg.com/browse/victory@36.6.8/es/index.js, so it should be picking up the re-exports from all those other packages.

Simmilarly, https://unpkg.com/browse/victory-area@36.6.8/package.json points to https://unpkg.com/browse/victory-area@36.6.8/es/index.js which points to https://unpkg.com/browse/victory-area@36.6.8/es/victory-area.js, so i'd expect this to work.

**Comment by azyzz228** [comment]
I was not able to reproduce the `import/name` error on my end. Error is not flagged when I turn on `import/named`.

```
> sandbox-new@1.0.0 debug
> npx eslint ./debug


/Users/azyzz/Documents/fellowship/sandbox-eslint-import/debug/victory-import.js
  1:10  error  'VictoryAxis' is defined but never used  no-unused-vars
  1:23  error  'VictoryLine' is defined but never used  no-unused-vars
  1:36  error  'VictoryArea' is defined but never used  no-unused-vars
```

 The repo of the project is [here](https://github.com/azyzz228/sandbox-eslint-import/tree/replicate-victory-import-error).

**Comment by ljharb** [comment]
@kaiyoma can you please provide your eslint config? Since we're unable to reproduce this, I'll close this for now, but am happy to reopen once we get more info.

**Comment by kaiyoma** [comment]
@ljharb <details><summary>Here's the eslint config:</summary>

```
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true
  },
  "globals": {
    "BigInt": "readonly",
    "EnvConfig": "writable",
    "jsdom": "readonly",
    "$Keys": "readonly",
    "$ReadOnlyArray": "readonly"
  },
  "parser": "C:\\Work\\event-viewer\\node_modules\\.pnpm\\@typescript-eslint+parser@5.44.0_s6zjlrzpskv7oc5fqmwkfkbdxm\\node_modules\\@typescript-eslint\\parser\\dist\\index.js",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "filenames",
    "import",
    "arista",
    "react-hooks",
    "react"
  ],
  "rules": {
    "filenames/match-exported": [
      2
    ],
    "react/forbid-prop-types": [
      "off"
    ],
    "react/default-props-match-prop-types": [
      "error"
    ],
    "react/jsx-boolean-value": [
      "error",
      "always"
    ],
    "react/jsx-filename-extension": [
      "error",
      {
        "allow": "as-needed",
        "extensions": [
          ".jsx",
          ".tsx"
        ]
      }
    ],
    "react/jsx-fragments": [
      "error"
    ],
    "react/jsx-key": [
      "error"
    ],
    "react/jsx-no-target-blank": [
      "error"
    ],
    "react/jsx-pascal-case": [
      "error"
    ],
    "react/jsx-sort-default-props": [
      "error"
    ],
    "react/jsx-uses-react": [
      "error"
    ],
    "react/jsx-uses-vars": [
      "error"
    ],
    "react/no-access-state-in-setstate": [
      "error"
    ],
    "react/no-deprecated": [
      "error"
    ],
    "react/no-direct-mutation-state": [
      "error"
    ],
    "react/no-find-dom-node": [
      "error"
    ],
    "react/no-string-refs": [
      "error"
    ],
    "react/no-typos": [
      "error"
    ],
    "react/no-unknown-property": [
      "error"
    ],
    "react/no-unsafe": [
      "error"
    ],
    "react/no-unused-state": [
      "error"
    ],
    "react/prop-types": [
      "error"
    ],
    "react/react-in-jsx-scope": [
      "error"
    ],
    "react/sort-prop-types": [
      "error",
      {
        "callbacksLast": true,
        "ignoreCase": false,
        "requiredFirst": true
      }
    ],
    "react-hooks/rules-of-hooks": [
      "error"
    ],
    "arista/import-order": [
      "error"
    ],
    "array-callback-return": [
      "error"
    ],
    "block-scoped-var": [
      "error"
    ],
    "consistent-return": [
      "error"
    ],
    "camelcase": [
      "error",
      {
        "allow": [
          "delete_all",
          "path_elements",
          "^UNSAFE_"
        ],
        "ignoreDestructuring": false,
        "ignoreImports": false,
        "ignoreGlobals": false
      }
    ],
    "constructor-super": [
      "error"
    ],
    "curly": [
      "error",
      "all"
    ],
    "default-case": [
      "error"
    ],
    "dot-notation": [
      "error"
    ],
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ],
    "for-direction": [
      "error"
    ],
    "func-names": [
      "error"
    ],
    "getter-return": [
      "error"
    ],
    "guard-for-in": [
      "error"
    ],
    "import/export": [
      "error"
    ],
    "import/first": [
      "error"
    ],
    "import/named": [
      "error"
    ],
    "import/no-absolute-path": [
      "error"
    ],
    "import/no-amd": [
      "error"
    ],
    "import/no-cycle": [
      "error"
    ],
    "import/no-deprecated": [
      "error"
    ],
    "import/no-duplicates": [
      "error"
    ],
    "import/no-dynamic-require": [
      "error"
    ],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "optionalDependencies": false
      }
    ],
    "import/no-mutable-exports": [
      "error"
    ],
    "import/no-named-as-default": [
      "error"
    ],
    "import/no-named-default": [
      "error"
    ],
    "import/no-self-import": [
      "error"
    ],
    "import/no-useless-path-segments": [
      "error",
      {
        "commonjs": true
      }
    ],
    "import/no-webpack-loader-syntax": [
      "error"
    ],
    "lines-between-class-members": [
      "error"
    ],
    "max-classes-per-file": [
      "error",
      1
    ],
    "max-len": [
      "error",
      {
        "code": 100,
        "ignorePattern": "^\\s+(it|test)\\(",
        "ignoreUrls": true
      }
    ],
    "new-cap": [
      "error",
      {
        "capIsNewExceptionPattern": "^Immutable",
        "newIsCap": true,
        "capIsNew": true,
        "properties": true
      }
    ],
    "no-alert": [
      "warn"
    ],
    "no-array-constructor": [
      "error"
    ],
    "no-async-promise-executor": [
      "error"
    ],
    "no-bitwise": [
      "error"
    ],
    "no-caller": [
      "error"
    ],
    "no-case-declarations": [
      "error"
    ],
    "no-class-assign": [
      "error"
    ],
    "no-compare-neg-zero": [
      "error"
    ],
    "no-cond-assign": [
      "error",
      "always"
    ],
    "no-confusing-arrow": [
      "error"
    ],
    "no-console": [
      "warn"
    ],
    "no-const-assign": [
      "error"
    ],
    "no-constant-condition": [
      "warn"
    ],
    "no-control-regex": [
      "error"
    ],
    "no-debugger": [
      "error"
    ],
    "no-delete-var": [
      "error"
    ],
    "no-dupe-args": [
      "error"
    ],
    "no-dupe-class-members": [
      "error"
    ],
    "no-dupe-keys": [
      "error"
    ],
    "no-duplicate-case": [
      "error"
    ],
    "no-else-return": [
      "error",
      {
        "allowElseIf": false
      }
    ],
    "no-empty-character-class": [
      "error"
    ],
    "no-empty-function": [
      "error",
      {
        "allow": [
          "arrowFunctions",
          "functions",
          "methods"
        ]
      }
    ],
    "no-empty-pattern": [
      "error"
    ],
    "no-empty": [
      "error"
    ],
    "no-eval": [
      "error"
    ],
    "no-ex-assign": [
      "error"
    ],
    "no-extend-native": [
      "error"
    ],
    "no-extra-bind": [
      "error"
    ],
    "no-extra-boolean-cast": [
      "error"
    ],
    "no-extra-label": [
      "error"
    ],
    "no-fallthrough": [
      "error"
    ],
    "no-func-assign": [
      "error"
    ],
    "no-global-assign": [
      "error"
    ],
    "no-implied-eval": [
      "error"
    ],
    "no-inner-declarations": [
      "error"
    ],
    "no-invalid-regexp": [
      "error"
    ],
    "no-iterator": [
      "error"
    ],
    "no-label-var": [
      "error"
    ],
    "no-labels": [
      "error"
    ],
    "no-lone-blocks": [
      "error"
    ],
    "no-lonely-if": [
      "error"
    ],
    "no-loop-func": [
      "error"
    ],
    "no-misleading-character-class": [
      "error"
    ],
    "no-mixed-operators": [
      "error",
      {
        "groups": [
          [
            "&",
            "|",
            "^",
            "~",
            "<<",
            ">>",
            ">>>"
          ],
          [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<="
          ],
          [
            "&&",
            "||"
          ],
          [
            "in",
            "instanceof"
          ]
        ],
        "allowSamePrecedence": true
      }
    ],
    "no-multi-assign": [
      "error"
    ],
    "no-multi-str": [
      "error"
    ],
    "no-nested-ternary": [
      "error"
    ],
    "no-new-func": [
      "error"
    ],
    "no-new-object": [
      "error"
    ],
    "no-new-symbol": [
      "error"
    ],
    "no-new-wrappers": [
      "error"
    ],
    "no-new": [
      "error"
    ],
    "no-obj-calls": [
      "error"
    ],
    "no-octal-escape": [
      "error"
    ],
    "no-octal": [
      "error"
    ],
    "no-param-reassign": [
      "error"
    ],
    "no-plusplus": [
      "error",
      {
        "allowForLoopAfterthoughts": true
      }
    ],
    "no-proto": [
      "error"
    ],
    "no-redeclare": [
      "error"
    ],
    "no-regex-spaces": [
      "error"
    ],
    "no-restricted-globals": [
      "error",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultstatus",
      "defaultStatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "isFinite",
      "isNaN",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top"
    ],
    "no-restricted-properties": [
      "error",
      {
        "property": "__defineGetter__",
        "message": "Use Object.defineProperty instead"
      },
      {
        "property": "__defineSetter__",
        "message": "Use Object.defineProperty instead"
      },
      {
        "object": "arguments",
        "property": "callee",
        "message": "arguments.callee is deprecated"
      },
      {
        "object": "global",
        "property": "isFinite",
        "message": "Use Number.isFinite instead"
      },
      {
        "object": "global",
        "property": "isNaN",
        "message": "Use Number.isNaN instead"
      },
      {
        "object": "self",
        "property": "isFinite",
        "message": "Use Number.isFinite instead"
      },
      {
        "object": "self",
        "property": "isNaN",
        "message": "Use Number.isNaN instead"
      },
      {
        "object": "window",
        "property": "isFinite",
        "message": "Use Number.isFinite instead"
      },
      {
        "object": "window",
        "property": "isNaN",
        "message": "Use Number.isNaN instead"
      },
      {
        "object": "Math",
        "property": "pow",
        "message": "Use the exponentiation operator (**) instead"
      }
    ],
    "no-return-assign": [
      "error",
      "always"
    ],
    "no-return-await": [
      "error"
    ],
    "no-script-url": [
      "error"
    ],
    "no-self-assign": [
      "error"
    ],
    "no-self-compare": [
      "error"
    ],
    "no-sequences": [
      "error"
    ],
    "no-shadow-restricted-names": [
      "error"
    ],
    "no-shadow": [
      "error"
    ],
    "no-sparse-arrays": [
      "error"
    ],
    "no-tabs": [
      "error"
    ],
    "no-template-curly-in-string": [
      "error"
    ],
    "no-this-before-super": [
      "error"
    ],
    "no-throw-literal": [
      "error"
    ],
    "no-undef-init": [
      "error"
    ],
    "no-undef": [
      "error"
    ],
    "no-underscore-dangle": [
      "error",
      {
        "allow": [
          "_key",
          "_value"
        ],
        "allowAfterThis": false,
        "allowAfterSuper": false,
        "allowAfterThisConstructor": false,
        "enforceInMethodNames": false,
        "allowFunctionParams": true,
        "enforceInClassFields": false
      }
    ],
    "no-unexpected-multiline": [
      "error"
    ],
    "no-unneeded-ternary": [
      "error",
      {
        "defaultAssignment": false
      }
    ],
    "no-unreachable": [
      "error"
    ],
    "no-unsafe-finally": [
      "error"
    ],
    "no-unsafe-negation": [
      "error"
    ],
    "no-unused-expressions": [
      "error"
    ],
    "no-unused-labels": [
      "error"
    ],
    "no-unused-vars": [
      "error",
      {
        "args": "after-used",
        "ignoreRestSiblings": true
      }
    ],
    "no-use-before-define": [
      "error"
    ],
    "no-useless-catch": [
      "error"
    ],
    "no-useless-computed-key": [
      "error"
    ],
    "no-useless-concat": [
      "error"
    ],
    "no-useless-constructor": [
      "error"
    ],
    "no-useless-escape": [
      "error"
    ],
    "no-useless-rename": [
      "error"
    ],
    "no-useless-return": [
      "error"
    ],
    "no-var": [
      "error"
    ],
    "no-void": [
      "error"
    ],
    "no-with": [
      "error"
    ],
    "object-curly-newline": [
      "error"
    ],
    "object-shorthand": [
      "error",
      "always",
      {
        "avoidQuotes": true,
        "ignoreConstructors": false
      }
    ],
    "one-var": [
      "error",
      "never"
    ],
    "operator-assignment": [
      "error",
      "always"
    ],
    "prefer-arrow-callback": [
      "error"
    ],
    "prefer-const": [
      "error",
      {
        "ignoreReadBeforeAssign": true,
        "destructuring": "any"
      }
    ],
    "prefer-numeric-literals": [
      "error"
    ],
    "prefer-promise-reject-errors": [
      "error",
      {
        "allowEmptyReject": true
      }
    ],
    "prefer-rest-params": [
      "error"
    ],
    "prefer-spread": [
      "error"
    ],
    "require-yield": [
      "error"
    ],
    "symbol-description": [
      "error"
    ],
    "radix": [
      "error"
    ],
    "use-isnan": [
      "error"
    ],
    "valid-typeof": [
      "error",
      {
        "requireStringLiterals": true
      }
    ],
    "vars-on-top": [
      "error"
    ],
    "yoda": [
      "error"
    ]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {},
      "webpack": {}
    }
  },
  "ignorePatterns": []
}
```

</details>

**Comment by ljharb** [comment]
@kaiyoma ok, you're definitely missing the `import/extensions` setting, as well as the typescript resolver, both of which are necessary if you're using TS.

**Comment by kaiyoma** [comment]
@ljharb This isn't a TS file. We're seeing this linting error in a JSX file.

**Comment by ljharb** [comment]
ok, you still need the `import/extensions` setting to include `.jsx` tho, and you definitely still need the TS resolver.
